packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -276,21 +277,50 @@ export class VerifiedFetch {
     return response
   }
 
+  // eslint-disable-next-line complexity
   private async handleDagPb ({ cid, path, resource, options }: FetchHandlerFunctionArg): Promise<Response> {
     let terminalElement: UnixFSEntry | undefined
     let ipfsRoots: CID[] | undefined
     let redirected = false
+    let offset: number | undefined
+    let length: number | undefined
+    let suffixLength: number | undefined
+    let isRangeRequest = false
+    try {
+      const rangeOptions = getRequestRange(options?.headers)
+      offset = rangeOptions?.offset
+      length = rangeOptions?.length
+      suffixLength = rangeOptions?.suffixLength
+      this.log.trace('range options %o', rangeOptions)
+      if (offset != null || length != null || suffixLength != null) {
+        isRangeRequest = true
+      }
+    } catch (err) {
+      this.log.error('Error parsing range request', err)
+      return badRangeResponse('Invalid range request')
+    }
 
     try {
-      const pathDetails = await walkPath(this.helia.blockstore, `${cid.toString()}/${path}`, options)
+      const pathDetails = await walkPath(this.helia.blockstore, `${cid.toString()}/${path}`, { ...options, offset, length })
```

Comment:
I'm not sure if we care about offset and length here, but I presumed so. If not, we can remove a lot of this code up here and potentially remove the `eslint-disable-next-line complexity`

Reply:
removed the offset & length here.